docs/src/conventions.md

```diff
@@ -101,8 +101,56 @@ based on the following rules.
   (or `wrap_array_native(u_ode, semi)`) for further processing.
 - When some solution is passed together with the `mesh, equations, solver, cache, ...`,
   it is already wrapped via `wrap_array` (or `wrap_array_native`).
-- Exceptions of this rule are possible, e.g. for AMR, but must be documented in
+- Exceptions of this rule are possible, e.g., for AMR, but must be documented in
   the code.
 - `wrap_array` should be used as default option. `wrap_array_native` should only
   be used when necessary, e.g., to avoid additional overhead when interfacing
   with external C libraries such as HDF5, MPI, or visualization.
+
+## Numeric types and type stability
+In Trixi.jl, `Float32` and `Float64` types are fully supported. We ensure the type stability of these numeric types throughout the development process. Below are some guidelines to apply in various scenarios.
+
+- **Exact floating-point numbers**: Some real numbers can be represented exactly in machine (e.g., `0.25`, `0.5`, `1/2`) and we prefer to use `Float32` type of them to achieve a concise way of possible type promotion. For example,
```

Comment:
If you want to be concrete, you need to require that `m` and `n` are close enough to zero, e.g.,
```julia
julia> Float32(2)^256
Inf32

julia> Float64(2)^256
1.157920892373162e77
```
and 
```julia
julia> Float32(2)^-256
0.0f0

julia> Float64(2)^-256
8.636168555094445e-78
```

Reply:
```suggestion
### Exact floating-point numbers

Some real numbers can be represented exactly as both `Float64` and `Float32` values (e.g., `0.25`, `0.5`, `1/2`). This is generally true for integer multiples of powers of two, i.e., numbers that can be written as ``m 2^n``, where ``m, n \in \mathbb{Z}`` and as long as ``m`` and ``n`` are not too large. We prefer to use `Float32` type for these numbers to achieve a concise way of possible type promotion. For example,
```
Good point. Maybe like this?

Reply:
Fine with me 👍

Reply:
I suggest not being too specific, instead, a general approach is preferred here. To be concrete, `m` and `n` should fall within a specific range, i.e., they should not be too large or too small. See
https://en.wikipedia.org/wiki/Single-precision_floating-point_format 
https://en.wikipedia.org/wiki/Double-precision_floating-point_format

Reply:
I agree. Being to specific here is not helpful. However, many constants used in Trixi.jl seem to be in the range of 0.001 to 10, thus giving a workable rule of thumb when a number can be represented exactly is imho very helpful to the developers. 

Since this convo is outdated, I'll make a new suggestion and resolve this one.